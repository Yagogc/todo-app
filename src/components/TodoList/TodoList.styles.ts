import styled, { css } from 'styled-components/macro'

const Ul = styled.ul`
  ${() => css`
    margin: 0;
    padding: 10px 0;
  `}
`

export { Ul }
