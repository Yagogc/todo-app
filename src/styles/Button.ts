import styled, { css } from 'styled-components/macro'

const Button = styled.button`
  ${() => css`
    padding: 8px;
    background-color: transparent;
    border: 1px solid var(--secondary-bg-color);

    &:hover {
      cursor: pointer;
    }
  `}
`

export { Button }
