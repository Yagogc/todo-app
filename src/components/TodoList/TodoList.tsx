import React, { useContext } from 'react'
import { TodoContext } from '../../state/TodoContext'
import TodoItem from '../TodoItem'
import { Ul } from './TodoList.styles'

const TodoList = () => {
  const { todos } = useContext(TodoContext)
  console.log('TodoList -> todos', todos)
  const todosIds = Object.keys(todos)
  if (todosIds.length === 0) {
    return <p>Add your first TODO</p>
  }
  return (
    <Ul>
      {todosIds.reverse().map((id) => {
        return <TodoItem id={id} key={id} />
      })}
    </Ul>
  )
}

export default TodoList
