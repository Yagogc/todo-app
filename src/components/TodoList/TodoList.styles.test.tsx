import React from 'react'
import { render } from '@testing-library/react'
import { Ul } from './TodoList.styles'

describe('ServiceTileContainer', () => {
  it('has default styles', () => {
    const { baseElement } = render(<Ul />)
    expect(baseElement).toMatchSnapshot()
  })
})
