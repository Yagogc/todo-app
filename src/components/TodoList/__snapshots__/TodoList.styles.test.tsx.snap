// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceTileContainer has default styles 1`] = `
<body>
  <div>
    <ul
      class="TodoListstyles__Ul-sc-10gobef-0 awKEB"
    />
  </div>
</body>
`;
